src/sign.rs

```diff
@@ -1,61 +1,249 @@
+            let sig = Signature::sign(&priv_key, &msg);
+            println!("Signature: {sig}");
+        }
+        // Sign with a private key interactively.
+        (None, None, true, None) => private_key_cli(data)?,
+        // TODO: If the user provides neither account or private flags, ask in interactive mode?
+        _ => bail!(
+            "Unexpected set of options passed to `forc wallet sign`.\n  \
+                 To sign with a wallet account, use `forc wallet sign --account <index> <data>`\n  \
+                 To sign with a private key, use `forc wallet sign --private <data>`",
+        ),
+    }
+    Ok(())
+}
+
+pub(crate) fn wallet_account_cli(wallet_path: &Path, account_ix: usize, data: Data) -> Result<()> {
+    let msg = msg_from_data(data)?;
+    sign_msg_with_wallet_account_cli(wallet_path, account_ix, &msg)
+}
+
+pub(crate) fn private_key_cli(data: Data) -> Result<()> {
+    match data {
+        Data::TxId { tx_id } => sign_msg_with_private_key_cli(&msg_from_hash32(tx_id))?,
+        Data::File { path } => sign_msg_with_private_key_cli(&msg_from_file(&path)?)?,
+        Data::Hex { hex_string } => sign_msg_with_private_key_cli(&msg_from_hex_str(&hex_string)?)?,
+        Data::String { string } => sign_msg_with_private_key_cli(&Message::new(string))?,
+    }
+    Ok(())
 }
 
-pub(crate) fn sign_transaction_with_private_key_cli(tx_id: Bytes32) -> Result<()> {
+fn sign_msg_with_private_key_cli(msg: &Message) -> Result<()> {
     let secret_key_input = prompt_password("Please enter the private key you wish to sign with: ")?;
-    let secret_key = SecretKey::from_str(&secret_key_input)?;
-    let signature = sign_transaction_with_private_key(tx_id, secret_key)?;
+    let signature = sign_with_private_key_str(msg, &secret_key_input)?;
     println!("Signature: {signature}");
     Ok(())
 }
 
-pub(crate) fn sign_transaction_cli(
+fn sign_with_private_key_str(msg: &Message, priv_key_input: &str) -> Result<Signature> {
+    let secret_key = SecretKey::from_str(priv_key_input)?;
+    Ok(Signature::sign(&secret_key, msg))
+}
+
+fn sign_msg_with_wallet_account_cli(
     wallet_path: &Path,
-    tx_id: Bytes32,
-    account_index: usize,
+    account_ix: usize,
+    msg: &Message,
 ) -> Result<()> {
     let password = prompt_password("Please enter your password: ")?;
-    let signature = sign_transaction(tx_id, account_index, &password, wallet_path)?;
+    let signature = sign_msg_with_wallet_account(wallet_path, account_ix, msg, &password)?;
     println!("Signature: {signature}");
     Ok(())
 }
 
+fn sign_msg_with_wallet_account(
+    wallet_path: &Path,
+    account_ix: usize,
+    msg: &Message,
+    pw: &str,
+) -> Result<Signature> {
+    let secret_key = account::derive_secret_key(wallet_path, account_ix, pw)?;
+    Ok(Signature::sign(&secret_key, msg))
+}
+
+// Cast the `Bytes32` directly to a message without normalizing it.
+// We don't renormalize as a hash is already a normalized representation.
```

Comment:
```suggestion
/// Cast the `Bytes32` directly to a message without normalizing it.
/// We don't renormalize as a hash is already a normalized representation.
```

Not important but maybe these were meant to be doc comments?